### PR TITLE
Adding special actions and clicks

### PR DESCRIPTION
**Pull request by MargotLepizzera**
### What does this PR do?
This PR adds documentation for browser tests:
- Special actions
- More type of clicks 
- New JS steps
- Included subtests (up to two levels of nesting for subtests)

### Motivation
New features

### Preview link
https://docs-staging.datadoghq.com/margot.lepizzera/synthetics_special_actions_clicks/synthetics/browser_tests/actions

### CODE REVIEW

File: content/en/synthetics/browser_tests/actions.md [hunk 1]
```diff
@@ -12,6 +12,19 @@ Actions are a series of steps that you can record for a browser test, which you
...
 
+## Click
+
+Datadog’s extension automatically records clicks on your page. 
+
+You can specify the type of click you want your browser test to perform at execution by selecting the click type of interest on your step:
+
+{{< img src="synthetics/browser_tests/browser_test_click_step.mp4" alt="Browser Test Click Step" video="true" width="60%">}}
+
+You can choose between:
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
Choose between:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 2]
```diff
@@ -39,12 +52,47 @@ The navigation action allows you to:
 * Go to an email and click on a link. This step allows you to access your Synthetics mail inbox after creating an [email variable][3].
 * Choose the email you are interested in and click the link you want your browser test to click on.
 
-## Hover
+## Special Actions
+
+Most of the actions are recorded using our extension. Some actions are however not recorded automatically and you therefore need to explicitly add a step about it using the **Special Actions** menu located at the top left hand corner of the recorder.
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]: We should put a link here to the extension:

```suggestion
Most of the actions are recorded using the provided Datadog extension. Some actions are however not recorded automatically and you need to explicitly add a step for them using the **Special Actions** menu located at the top left hand corner of the recorder.
```

File: content/en/synthetics/browser_tests/actions.md [hunk 3]
```diff
@@ -39,12 +52,47 @@ The navigation action allows you to:
...
+
+### Hover
 
 This browser test step isn’t added through an actual hovering mechanism (otherwise each element you are hovering would be added as a step) but using a dedicated action with a click.
 
 After selecting the Hover action, click on the element you want to choose to create a new step.
 
+### Press Key
+
+You can simulate users entering keystrokes using **Press Key** steps. The below keys can be recorded using our extension:
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
You can simulate users entering keystrokes using **Press Key** steps. The keys below can be recorded using Datadog extension:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 4]
```diff
@@ -39,12 +52,47 @@ The navigation action allows you to:
...
+
+* Enter
+* Arrows
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
* Arrows (up, down, right, and left)
```

File: content/en/synthetics/browser_tests/actions.md [hunk 5]
```diff
@@ -39,12 +52,47 @@ The navigation action allows you to:
...
+* Tab (outside a form)
+* Escape 
+* Backspace
+
+If you need your browser test to press keys that are not being automatically recorded, you can leverage the **Press Key** special action and specify which values need to be pressed in the value box.
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
If you need your browser test to press keys that are not being automatically recorded, specify which values need to be pressed in the value box of the  **Press Key** special action:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 6]
```diff
@@ -39,12 +52,47 @@ The navigation action allows you to:
...
+
+### Scroll
+
+Your browser test automatically scrolls to the element it needs to interact with. In most cases you consequently do not need to add a manual scroll. The scroll step should only be added when needed to trigger an additional network request such as an infinite scroll.
+
+{{< img src="synthetics/browser_tests/browser_test_scroll_step.png" alt="Browser Test Scroll Step"  style="width:60%;">}}
+
+You need to specify the number of pixels your browser test should scroll vertically or horizontally.
+
+By default, our scroll step will scroll on the whole page. If you however need to scroll a specific element (for instance a specific div), you can leverage the **Target Element** option in order to select the element you want your browser test to scroll at test execution.
```
Comment (l0k0ms) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
By default, the test scroll step scrolls on the whole page. If you need to scroll to a specific element (for instance a specific `<div>`), use the **Target Element** option to select the element you want your browser test to scroll at.
```

File: content/en/synthetics/browser_tests/actions.md [hunk 7]
```diff
@@ -12,6 +12,19 @@ Actions are a series of steps that you can record for a browser test, which you
 
 **Note**: The default timeout for each step is approximately 60 seconds. You can override the timeout with [advanced options][2].
 
+## Click
+
+[Datadog’s extension][4] automatically records clicks on your page. 
+
+You can specify the type of click you want your browser test to perform at execution by selecting the click type of interest on your step:
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
Specify the type of click you want your browser test to perform at execution time:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 8]
```diff
@@ -12,6 +12,19 @@ Actions are a series of steps that you can record for a browser test, which you
...
+
+{{< img src="synthetics/browser_tests/browser_test_click_step.mp4" alt="Browser Test Click Step" video="true" width="60%">}}
+
+Choose between:
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
Choose from:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 9]
```diff
@@ -12,6 +12,19 @@ Actions are a series of steps that you can record for a browser test, which you
...
+
+* Primary click (left) which corresponds to a left click
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
* Primary click (corresponds to a left click)
```

File: content/en/synthetics/browser_tests/actions.md [hunk 10]
```diff
@@ -12,6 +12,19 @@ Actions are a series of steps that you can record for a browser test, which you
...
+* Double click
+* Contextual click (right click)
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
* Contextual click (corresponds to a right click)
```

File: content/en/synthetics/browser_tests/actions.md [hunk 11]
```diff
@@ -39,12 +53,47 @@ The navigation action allows you to:
 * Go to an email and click on a link. This step allows you to access your Synthetics mail inbox after creating an [email variable][3].
 * Choose the email you are interested in and click the link you want your browser test to click on.
 
-## Hover
+## Special Actions
+
+Most of the actions are recorded using [the provided Datadog extension][4]. Some actions are however not recorded automatically and you need to explicitly add a step for them using the **Special Actions** menu located at the top left-hand corner of the recorder.
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
[The provided Datadog extension][4] records most actions. However, some actions—such as hover, press key, and scroll—are not recorded automatically. Explicitly add a step for them using the **Special Actions** menu located at the top left-hand corner of the recorder.
```

File: content/en/synthetics/browser_tests/actions.md [hunk 12]
```diff
@@ -39,12 +53,47 @@ The navigation action allows you to:
...
+
+### Hover
 
 This browser test step isn’t added through an actual hovering mechanism (otherwise each element you are hovering would be added as a step) but using a dedicated action with a click.
 
 After selecting the Hover action, click on the element you want to choose to create a new step.
 
+### Press Key
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
### Press key
```

File: content/en/synthetics/browser_tests/actions.md [hunk 13]
```diff
@@ -39,12 +53,47 @@ The navigation action allows you to:
...
+
+You can simulate users entering keystrokes using **Press Key** steps. The keys below can be recorded using [Datadog extension][4]:
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
You can simulate users entering keystrokes using **Press Key** steps. The keys below can be recorded using the [Datadog extension][4]:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 14]
```diff
@@ -39,12 +53,47 @@ The navigation action allows you to:
...
+
+* Enter
+* Arrows (up, down, right, and left)
+* Tab (outside a form)
+* Escape 
+* Backspace
+
+To press keys that are not being automatically recorded, specify which values need to be pressed in the value box of the  **Press Key** special action:
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
To press keys that are not being automatically recorded, specify which values need to be pressed in the value box of the **Press Key** special action:
```

File: content/en/synthetics/browser_tests/actions.md [hunk 15]
```diff
@@ -39,12 +53,47 @@ The navigation action allows you to:
...
+The below modifiers can also be applied to the inputted value:
+
+* Alt
+* Control
+* Meta
+* Shift
+
+### Scroll
+
+Your browser test automatically scrolls to the element it needs to interact with. In most cases you consequently do not need to add a manual scroll. The scroll step should only be added when needed to trigger an additional network request such as an infinite scroll.
```
Comment (cswatt) [inline content/en/synthetics/browser_tests/actions.md]:
```suggestion
Your browser test automatically scrolls to the element it needs to interact with. Consequently, in most cases, you do not need to manually add a scroll step. The scroll step should only be added when needed to trigger an additional network request, such as in an infinite scroll.
```

**Pull request merged**